common/frame.go

```diff
@@ -1849,17 +1849,19 @@ func (f *Frame) WaitForNavigation(opts goja.Value) *goja.Promise {
 			return false
 		})
 
-	handleTimeoutError := func(err error) {
-		if errors.Is(err, context.DeadlineExceeded) {
-			err = &k6ext.UserFriendlyError{
+	handleTimeoutError := func(err error) error {
+		if err != nil {
+			e := &k6ext.UserFriendlyError{
 				Err:     err,
 				Timeout: parsedOpts.Timeout,
 			}
-			k6ext.Panic(f.ctx, "waiting for navigation: %w", err)
+			return fmt.Errorf("waiting for navigation: %w", e)
 		}
 		f.log.Debugf("Frame:WaitForNavigation",
 			"fid:%v furl:%s timeoutCtx done: %v",
 			f.ID(), f.URL(), err)
+
+		return nil
 	}
```

Comment:
Might be confusing if the error is not `context.DeadlineExceeded`, although i realise that this lambda func is only used when the context is closed. I would also move the debug log line to just before returning the wrapped error. All minor things though in this case.

Reply:
Since 65fbc97, this will now always return an error if it's not nil, so this `return` likely won't be reached.

The debug logging was here to avoid a double log of both the timeout err and this debug message, but I guess it doesn't make sense to have it if `err` will be nil here. And double logging is not a big deal, so I moved it in 5751ddc.